include/openmc/random_ray/tally_convert.h

```diff
@@ -0,0 +1,35 @@
+#ifndef OPENMC_RANDOM_RAY_TALLY_CONVERT_H
+#define OPENMC_RANDOM_RAY_TALLY_CONVERT_H
+
+#include <vector>
+
+namespace openmc {
+
+// A mapping object that is used to map between a specific random ray
+// source region and an OpenMC native tally bin that it should score to
+// every iteration.
+struct TallyTask {
+  int tally_idx;
+  int filter_idx;
+  int score_idx;
+  int score_type;
+  TallyTask(int tally_idx, int filter_idx, int score_idx, int score_type)
+    : tally_idx(tally_idx), filter_idx(filter_idx), score_idx(score_idx),
+      score_type(score_type)
+  {}
+};
+
+namespace random_ray {
+
+// 2D array representing values for all source regions x energy groups x tally
+// tasks
+extern std::vector<std::vector<TallyTask>> tally_task;
```

Comment:
Is the inner dimension the same for each? It seems to be the case, so I suggest using a single vector with the index computed by the inner dimension. Could put a small struct wrapper around it with `operator(int i_source, int i_group)` defined if you want to retain syntax like `tally_task[i_source, i_group]`. I have a class called `RectangularArray` I wrote for some other code that does exactly this. Maybe there are some other places it'd make sense to have.

Reply:
The inner dimension I think is variable length depending on the FSR. Depending on how the user defines their tallies, some FSRs may score to multiple tallies, while others may score to none, creating the need for a jagged 2D array here.

E.g., in a reactor core simulation, a user might want pin-wise tallies in assembly A as well as a tally over the full assembly A, meanwhile assemblies B, C, D etc have no tallies associated with them at all.